.github/workflows/docker-build-push-ecr.yaml

```diff
@@ -52,6 +58,12 @@ jobs:
       - name: Check out code
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
 
+      - name: Download artifacts to pass to docker build
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4
+        with:
+          name: ${{ inputs.artifact_name }}
+          path: ${{ inputs.artifact_path }}
```

Comment:
You probably need some conditions here as the artifact inputs are not required. Did you test passing empty values ?

Reply:
Thanks @Davidoutz for making me reason twice about this. 

It was working, but it's not what we want. If you don't pass values it will download all artifacts by default and that might pollute the docker context. I set to download artifacts only when specifying the path or the name.